include/flyft/data_layout.h

```diff
@@ -3,24 +3,39 @@
 
 #include <iterator>
 #include <type_traits>
+#include <vector>
 
 namespace flyft
     {
+//! DataLayout
+/*!
+ * Generates data structure for the custom data types used in the software.
+ * The class provides get and set functions for the array index and implements N-dimensional array
+ * mapping into one dimensional array using strided indexing scheme.
+ */
 class DataLayout
     {
     public:
     DataLayout();
-    explicit DataLayout(int shape_);
+    //! Constructor for the DataLayout
+    /*!
+     * \param shape_ Shape of the index array
+     */
+    explicit DataLayout(const std::vector<int>& shape);
 
-    int operator()(int idx) const;
-    int shape() const;
+    // Getter for the class
+    int operator()(const std::vector<int>& idx) const;
```

Comment:
This is not a doxygen comment and not descriptive
```suggestion
    //! Map a multidimensional index to a one-dimensional index
    /*!
     * \param idx Multidimensional index.
     * \return One-dimensional index.
     *
     * Row-major ordering is used.
     */
    int operator()(const std::vector<int>& idx) const;
```